examples/flight-animation.js

```diff
@@ -0,0 +1,135 @@
+// NOCOMPILE
+// this example uses arc.js for which we don't have an externs file.
+goog.require('ol.Attribution');
+goog.require('ol.Feature');
+goog.require('ol.Map');
+goog.require('ol.View');
+goog.require('ol.control');
+goog.require('ol.geom.LineString');
+goog.require('ol.layer.Tile');
+goog.require('ol.layer.Vector');
+goog.require('ol.proj');
+goog.require('ol.source.Stamen');
+goog.require('ol.source.Vector');
+goog.require('ol.style.Stroke');
+goog.require('ol.style.Style');
+
+var map = new ol.Map({
+  layers: [
+    new ol.layer.Tile({
+      source: new ol.source.Stamen({
+        layer: 'toner'
+      })
+    })
+  ],
+  controls: ol.control.defaults({
+    attributionOptions: /** @type {olx.control.AttributionOptions} */ ({
+      collapsible: false
+    })
+  }),
+  renderer: 'canvas',
+  target: 'map',
+  view: new ol.View({
+    center: [0, 0],
+    zoom: 2
+  })
+});
+
+var defaultStroke = new ol.style.Stroke({
+  color: '#EAE911',
+  width: 2
+});
+var defaultStyle = new ol.style.Style({
+  stroke: defaultStroke
+});
+
+var pointsPerMs = 0.1;
+var animateFlights = function(event) {
+  var vectorContext = event.vectorContext;
+  var frameState = event.frameState;
+  vectorContext.setFillStrokeStyle(null, defaultStroke);
+
+  var features = flightsSource.getFeatures();
+  for (var i = 0; i < features.length; i++) {
+    var feature = features[i];
+    if (!feature.get('finished')) {
+      // only draw the lines for which the animation has not
+      // finished yet
+      var coords = feature.getGeometry().getCoordinates();
+      var elapsedTime = frameState.time - feature.get('start');
+      var elapsedPoints = elapsedTime * pointsPerMs;
+
+      if (elapsedPoints >= coords.length) {
+        feature.set('finished', true);
+      }
+
+      var maxIndex = Math.min(elapsedPoints, coords.length);
+      var currentLine = new ol.geom.LineString(coords.slice(0, maxIndex));
+
+      // directly draw the line with the vector context
+      vectorContext.drawLineStringGeometry(currentLine, feature);
+    }
+  }
+  // tell OL3 to continue the postcompose animation
+  frameState.animate = true;
+};
+
+var addLater = function(feature, timeout) {
+  window.setTimeout(function() {
+    feature.set('start', new Date().getTime());
+    flightsSource.addFeature(feature);
+  }, timeout);
+};
+
+var flightsSource = new ol.source.Vector({
+  wrapX: false,
+  attributions: [new ol.Attribution({
+    html: 'Flight data by ' +
+        '<a href="http://openflights.org/data.html">OpenFlights</a>,'
+  })],
+  loader: function(extent, resolution, projection) {
+    var url = 'data/openflights/flights.json';
+    $.ajax({url: url, dataType: 'json', success: function(response) {
```

Comment:
After #4420 is complete, examples that use jQuery will need to have it in the `resources` list in the YAML front-matter.

I'd like to limit the use of jQuery to examples that demonstrate integration with a jQuery plugin (e.g. the tooltip/popover examples).  Instead of `$.ajax()`, I'm converting examples to use [`fetch()`](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) (with a polyfill for IE/Safari).

No need to change this here yet, but just a heads up on what future examples will use.